File: examples/api-client.py
"""
CodeContext Live API Client - Python
Simple Python client for interacting with CodeContext Live API
"""

import requests
from typing import List, Dict, Optional
import json


class CodeContextClient:
    """Client for CodeContext Live API"""

    def __init__(self, base_url: str = "http://localhost:3000"):
        """
        Initialize the client

        Args:
            base_url: Base URL of the CodeContext Live server
        """
        self.base_url = base_url
        self.session = requests.Session()
        self.session.headers.update({"Content-Type": "application/json"})

    def get_context(self, file_path: str) -> Dict:
        """
        Get structured context for a file

        Args:
            file_path: Path to the file

        Returns:
            Context digest dictionary
        """
        response = self.session.post(
            f"{self.base_url}/api/context",
            json={"filePath": file_path}
        )
        response.raise_for_status()
        return response.json()["data"]

    def get_context_markdown(self, file_path: str) -> str:
        """
        Get Markdown-formatted context summary

        Args:
            file_path: Path to the file

        Returns:
            Markdown summary string
        """
        response = self.session.post(
            f"{self.base_url}/api/context/markdown",
            json={"filePath": file_path}
        )
        response.raise_for_status()
        return response.json()["data"]

    def analyze_file(self, file_path: str) -> str:
        """
        Manually trigger analysis for a file

        Args:
            file_path: Path to the file

        Returns:
            Markdown summary string
        """
        response = self.session.post(
            f"{self.base_url}/api/analyze",
            json={"filePath": file_path}
        )
        response.raise_for_status()
        return response.json()["data"]

    def search(self, query: str, top_k: int = 5) -> List[Dict]:
        """
        Search the vector store

        Args:
            query: Search query
            top_k: Number of results to return

        Returns:
            List of search results
        """
        response = self.session.post(
            f"{self.base_url}/api/search",
            json={"query": query, "topK": top_k}
        )
        response.raise_for_status()
        return response.json()["data"]

    def index_codebase(self, directories: List[str]) -> Dict:
        """
        Index codebase directories

        Args:
            directories: List of directory paths to index

        Returns:
            Indexing status
        """
        response = self.session.post(
            f"{self.base_url}/api/index",
            json={"directories": directories}
        )
        response.raise_for_status()
        return response.json()

    def get_config(self) -> Dict:
        """
        Get server configuration

        Returns:
            Configuration dictionary
        """
        response = self.session.get(f"{self.base_url}/api/config")
        response.raise_for_status()
        return response.json()["data"]

    def health_check(self) -> Dict:
        """
        Check server health

        Returns:
            Health status dictionary
        """
        response = self.session.get(f"{self.base_url}/health")
        response.raise_for_status()
        return response.json()


def main():
    """Example usage"""
    client = CodeContextClient("http://localhost:3000")

    try:
        # Health check
        print("Checking server health...")
        health = client.health_check()
        print(f"Server status: {health['status']}")

        # Get configuration
        print("\nGetting configuration...")
        config = client.get_config()
        print(f"Config: {json.dumps(config, indent=2)}")

        # Analyze a file
        print("\nAnalyzing file...")
        file_path = "./examples/sample-files/SCR100.cbl"
        markdown = client.get_context_markdown(file_path)
        print("\n" + markdown)

        # Search for related code
        print("\nSearching for related code...")
        results = client.search("customer update", 3)
        print(f"Found {len(results)} related files")
        for i, result in enumerate(results, 1):
            print(f"{i}. {result['metadata']['fileName']}")

    except requests.exceptions.RequestException as e:
        print(f"Error: {e}")


if __name__ == "__main__":
    main()

File: examples/api-client.js
/**
 * CodeContext Live API Client
 * Simple JavaScript client for interacting with CodeContext Live API
 */

const axios = require('axios');

class CodeContextClient {
  constructor(baseUrl = 'http://localhost:3000') {
    this.baseUrl = baseUrl;
    this.client = axios.create({
      baseURL: baseUrl,
      timeout: 30000,
      headers: {
        'Content-Type': 'application/json',
      },
    });
  }

  /**
   * Get structured context for a file
   * @param {string} filePath - Path to the file
   * @returns {Promise<Object>} Context digest
   */
  async getContext(filePath) {
    try {
      const response = await this.client.post('/api/context', { filePath });
      return response.data.data;
    } catch (error) {
      throw this.handleError(error);
    }
  }

  /**
   * Get Markdown-formatted context summary
   * @param {string} filePath - Path to the file
   * @returns {Promise<string>} Markdown summary
   */
  async getContextMarkdown(filePath) {
    try {
      const response = await this.client.post('/api/context/markdown', { filePath });
      return response.data.data;
    } catch (error) {
      throw this.handleError(error);
    }
  }

  /**
   * Manually trigger analysis for a file
   * @param {string} filePath - Path to the file
   * @returns {Promise<string>} Markdown summary
   */
  async analyzeFile(filePath) {
    try {
      const response = await this.client.post('/api/analyze', { filePath });
      return response.data.data;
    } catch (error) {
      throw this.handleError(error);
    }
  }

  /**
   * Search the vector store
   * @param {string} query - Search query
   * @param {number} topK - Number of results
   * @returns {Promise<Array>} Search results
   */
  async search(query, topK = 5) {
    try {
      const response = await this.client.post('/api/search', { query, topK });
      return response.data.data;
    } catch (error) {
      throw this.handleError(error);
    }
  }

  /**
   * Index a codebase directory
   * @param {Array<string>} directories - Directories to index
   * @returns {Promise<Object>} Indexing status
   */
  async indexCodebase(directories) {
    try {
      const response = await this.client.post('/api/index', { directories });
      return response.data;
    } catch (error) {
      throw this.handleError(error);
    }
  }

  /**
   * Get server configuration
   * @returns {Promise<Object>} Configuration
   */
  async getConfig() {
    try {
      const response = await this.client.get('/api/config');
      return response.data.data;
    } catch (error) {
      throw this.handleError(error);
    }
  }

  /**
   * Check server health
   * @returns {Promise<Object>} Health status
   */
  async healthCheck() {
    try {
      const response = await this.client.get('/health');
      return response.data;
    } catch (error) {
      throw this.handleError(error);
    }
  }

  handleError(error) {
    if (error.response) {
      return new Error(`API Error: ${error.response.data.error || error.message}`);
    } else if (error.request) {
      return new Error('No response from server. Is CodeContext Live running?');
    } else {
      return new Error(`Request Error: ${error.message}`);
    }
  }
}

// Example usage
async function main() {
  const client = new CodeContextClient('http://localhost:3000');

  try {
    // Health check
    console.log('Checking server health...');
    const health = await client.healthCheck();
    console.log('Server status:', health.status);

    // Get configuration
    console.log('\nGetting configuration...');
    const config = await client.getConfig();
    console.log('Config:', JSON.stringify(config, null, 2));

    // Analyze a file
    console.log('\nAnalyzing file...');
    const filePath = './examples/sample-files/SCR100.cbl';
    const markdown = await client.getContextMarkdown(filePath);
    console.log('\n' + markdown);

    // Search for related code
    console.log('\nSearching for related code...');
    const results = await client.search('customer update', 3);
    console.log('Found', results.length, 'related files');
    results.forEach((result, index) => {
      console.log(`${index + 1}. ${result.metadata.fileName}`);
    });
  } catch (error) {
    console.error('Error:', error.message);
  }
}

// Run example if executed directly
if (require.main === module) {
  main();
}

module.exports = CodeContextClient;
